src/tasks/cosplay/array/list/ArrayNotList.java

```diff
@@ -0,0 +1,129 @@
+package tasks.cosplay.array.list;
+
+import java.util.Arrays;
+
+public class ArrayNotList<T> implements NotList<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int indexForNextPut = 0;
+
+    public ArrayNotList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public int getIndexForNextPut() {
+        return indexForNextPut;
+    }
+
+    public void arrayGrower() {
+        T[] temp;
```

Comment:
try  to simplify:
T[] temp = (T[]) new Object[array.length + (array.length >> 1)];

better move (array.length + (array.length >> 1)) in separate variable